pkg/test/verify/cluster.go

```diff
@@ -25,46 +25,41 @@ func AddToolchainClusterAsMember(t *testing.T, functionToVerify FunctionToVerify
 	defer gock.Off()
 	status := test.NewClusterStatus(toolchainv1alpha1.ToolchainClusterReady, corev1.ConditionTrue)
 	memberLabels := []map[string]string{
-		Labels("", "", test.NameHost),
-		Labels(cluster.Member, "", test.NameHost),
-		Labels(cluster.Member, "member-ns", test.NameHost)}
-	for _, labels := range memberLabels {
+		Labels("", test.NameHost),
+		Labels("member-ns", test.NameHost)}
 
+	for _, labels := range memberLabels {
 		t.Run("add member ToolchainCluster", func(t *testing.T) {
 			for _, withCA := range []bool{true, false} {
-				toolchainCluster, sec := test.NewToolchainCluster("east", "secret", status, labels)
+				toolchainCluster, sec := test.NewToolchainCluster("east", test.HostOperatorNs, "secret", status, labels)
 				if withCA {
 					toolchainCluster.Spec.CABundle = "ZHVtbXk="
 				}
+				toolchainCluster.Labels[cluster.RoleLabel(cluster.Tenant)] = ""
 				cl := test.NewFakeClient(t, toolchainCluster, sec)
 				service := newToolchainClusterService(t, cl, withCA)
 				defer service.DeleteToolchainCluster("east")
-
 				// when
 				err := functionToVerify(toolchainCluster, cl, service)
-
 				// then
-				require.NoError(t, err)
-				cachedToolchainCluster, ok := cluster.GetCachedToolchainCluster("east")
-				require.True(t, ok)
-				assert.Equal(t, cluster.Member, cachedToolchainCluster.Type)
-				if labels["namespace"] == "" {
-					assert.Equal(t, "toolchain-member-operator", cachedToolchainCluster.OperatorNamespace)
-				} else {
+				if labels["namespace"] != "" {
```

Comment:
I'd suggest to split this test in two as it is actually checking 2 different behaviors. Something like:

```go
	t.Run("add member ToolchainCluster with namespace label set", func(t *testing.T) {
		labels := Labels("", test.NameHost)

		for _, withCA := range []bool{true, false} {
			toolchainCluster, sec := test.NewToolchainCluster("east", test.HostOperatorNs, "secret", status, labels)
			if withCA {
				toolchainCluster.Spec.CABundle = "ZHVtbXk="
			}
			toolchainCluster.Labels[cluster.RoleLabel(cluster.Tenant)] = ""
			cl := test.NewFakeClient(t, toolchainCluster, sec)
			service := newToolchainClusterService(t, cl, withCA)
			defer service.DeleteToolchainCluster("east")

			// when
			err := functionToVerify(toolchainCluster, cl, service)

			// then
			require.NoError(t, err)
			cachedToolchainCluster, ok := cluster.GetCachedToolchainCluster("east")
			require.True(t, ok)
			assert.Equal(t, labels["namespace"], cachedToolchainCluster.OperatorNamespace)
			// check that toolchain cluster role label tenant was set only on member cluster type
			require.NoError(t, cl.Get(context.TODO(), client.ObjectKeyFromObject(toolchainCluster), toolchainCluster))
			_, found := toolchainCluster.Labels[cluster.RoleLabel(cluster.Tenant)]
			require.True(t, found)

			assert.Equal(t, status, *cachedToolchainCluster.ClusterStatus)
			assert.Equal(t, test.NameHost, cachedToolchainCluster.OwnerClusterName)
			assert.Equal(t, "http://cluster.com", cachedToolchainCluster.APIEndpoint)
		}
	})

	t.Run("add member ToolchainCluster without namespace label should fail", func(t *testing.T) {
		labels := Labels("member-ns", test.NameHost)

		for _, withCA := range []bool{true, false} {
			toolchainCluster, sec := test.NewToolchainCluster("east", test.HostOperatorNs, "secret", status, labels)
			if withCA {
				toolchainCluster.Spec.CABundle = "ZHVtbXk="
			}
			toolchainCluster.Labels[cluster.RoleLabel(cluster.Tenant)] = ""
			cl := test.NewFakeClient(t, toolchainCluster, sec)
			service := newToolchainClusterService(t, cl, withCA)
			defer service.DeleteToolchainCluster("east")
			// when
			err := functionToVerify(toolchainCluster, cl, service)
			// then
			require.Error(t, err)
			_, ok := cluster.GetCachedToolchainCluster("east")
			require.False(t, ok)
		}
	})
```

Reply:
updated it